doc/configuration.md

```diff
@@ -646,24 +644,24 @@ available:
 
 By default PHARs are `SHA1` signed.
 
+The `OPENSSL` algorithm will require to provide [a key][key]. 
 
-### The private key (`key`)
 
-// TODO: review this setting + doc, default value...]
+### The private key (`key`)
 
-The key (`string`|`null` default none) setting is used to specify the path to the private key file. The private key file
+The key (`string`|`null` default `null`) setting is used to specify the path to the private key file. The private key file
 will be used to sign the PHAR using the `OPENSSL` signature algorithm (see [Signing algorithm][algorithm]) and the
 setting will be completely ignored otherwise. If an absolute path is not provided, the path will be relative to the
 current working directory.
 
 
 ### The private key password (`key-pass`)
 
-// TODO: review this setting + doc, default value...]
-
-The private key password  (`string`|`boolean`|`null` default none) setting is used to specify the pass-phrase for the
+The private key password  (`string`|`boolean`|`null` default `null`) setting is used to specify the pass-phrase for the
 private key. If a string is provided, it will be used as is as the pass-phrase. If `true` is provided, you will be
-prompted for the passphrase.
+prompted for the passphrase unless you are not in an interactive environment.
+
+This setting will be ignored if no [key][key] has been provided.
```

Comment:
`[key][key]` ?
what this MD syntax stands for ?

Reply:
Instead of doing:

```md
[link text](link)
```

You can do:

```md
[link text][anchor]

[anchor]: link
```

Which allows you to group all the links in one place (here at the bot of the file) and re-use a given link in several places